sbi/neural_nets/density_estimators/base.py

```diff
@@ -1,23 +1,27 @@
+from typing import Optional, Tuple
+
 import torch
 from torch import Tensor, nn
 
 
-class DensityEstimator:
+class DensityEstimator(nn.Module):
     r"""Base class for density estimators.
 
     The density estimator class is a wrapper around neural networks that
     allows to evaluate the `log_prob`, `sample`, and provide the `loss` of $theta,x$
     pairs.
     """
 
-    def __init__(self, net: nn.Module) -> None:
+    def __init__(self, net: nn.Module, x_shape: torch.Size) -> None:
```

Comment:
maybe we want to name this `event_shape` because the context could also be `theta`?

Reply:
or `condition_shape`, or `c_shape`?

Reply:
Fine, with that (choose x_shape to make it consistent with the other class, but yeah for NLE we will have theta not x).

I like condition_shape, will change it to that.